File: backend/src/main/java/arbremojo/backend/model/user/Professor.java
package arbremojo.backend.model.user;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "professor")
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Professor extends User{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "professor_id", nullable = false)
    private Integer professor_id;
    @Column(name = "first_name", nullable = false)
    private String firstName;
    @Column(name = "last_name", nullable = false)
    private String lastName;
    @Column(name = "email", nullable = false)
    private String email;
    @Column(name = "password", nullable = false)
    private String password;
    @Column(name = "token")
    private String token;
    @Column(name = "pfp_img_path")
    private String pfpImgPath;

    @Override
    public int getUserId() {
        return professor_id;
    }
}

File: frontend/src/model/user/professor.ts
import {User} from "./user";

export class Professor extends User {
  constructor(firstName: string, lastName: string,
              email: string, password: string,
              professorId?: number | undefined,
              token?: string | undefined,
              pfpImgPath?: string | undefined,
              pfpImgUrl?: string | undefined) {
    super(firstName, lastName, email, password, professorId, token, pfpImgPath, pfpImgUrl);
    this.professorId = professorId;
  }

  static override fromJson(jsonProfessor: Professor): Professor {
    return new Professor(jsonProfessor.firstName, jsonProfessor.lastName,
      jsonProfessor.email, jsonProfessor.password, jsonProfessor.professorId, jsonProfessor.token, jsonProfessor.pfpImgPath,
      jsonProfessor.pfpImgUrl);
  }
}
